integrations/Matomo/browser.js

```diff
@@ -0,0 +1,205 @@
+        window._paq.push(["enableLinkTracking", true]);
+      }
+
+      // Disables page performance tracking.
+      if (this.disablePerformanceTracking) {
+        window._paq.push(["disablePerformanceTracking"]);
+      }
+
+      // Enables cross domain linking. It is useful if you own multiple domains and would like to track all the actions and
+      // pageviews of a specific visitor into the same visit.
+      if (this.enableCrossDomainLinking) {
+        window._paq.push(["enableCrossDomainLinking"]);
+      }
+
+      // Sets the cross domain linking timeout (in seconds).
+      if (this.setCrossDomainLinkingTimeout && this.timeout) {
+        window._paq.push(["setCrossDomainLinkingTimeout", this.timeout]);
+      }
+
+      // Gets the query parameter to append to the links to handle cross domain linking.
+      if (this.getCrossDomainLinkingUrlParameter) {
+        window._paq.push(["getCrossDomainLinkingUrlParameter"]);
+      }
+
+      // Disables the browser feature detection.
+      if (this.disableBrowserFeatureDetection) {
+        window._paq.push(["disableBrowserFeatureDetection"]);
+      }
+    }
+    return false;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Matomo Identify===");
+    const { anonymousId, userId } = rudderElement.message;
+    const matomoUserId = userId || anonymousId;
+    if (!matomoUserId) {
+      logger.error(
+        "User parameter (anonymousId or userId) is required for identify call"
+      );
+      return;
+    }
+    window._paq.push(["setUserId", matomoUserId]);
+  }
+
+  track(rudderElement) {
+    logger.debug("===In Matomo track===");
+
+    const { message } = rudderElement;
+    const { event } = message;
+    const goalListMap = getHashFromArrayWithDuplicate(this.eventsMapToGoalId);
+    const standardEventsMap = getHashFromArrayWithDuplicate(
+      this.eventsToStandard
+    );
+    const ecommerceMapping = new Map([
+      ["product viewed", this.ecomEvents.SET_ECOMMERCE_VIEW],
+      ["product added", this.ecomEvents.ADD_ECOMMERCE_ITEM],
+      ["product removed", this.ecomEvents.REMOVE_ECOMMERCE_ITEM],
+      ["order completed", this.ecomEvents.TRACK_ECOMMERCE_ORDER],
+      ["cart cleared", this.ecomEvents.CLEAR_ECOMMERCE_CART],
+      ["update cart", this.ecomEvents.TRACK_ECOMMERCE_CART_UPDATE],
+    ]);
+
+    if (!event) {
+      logger.error("Event name not present");
+      return;
+    }
+
+    // Checks for custom dimensions in the payload, if present makes appropriate calls
+    checkCustomDimensions(message);
+
+    // For every type of track calls we consider the trackGoal function.
+    if (goalListMap[event.toLowerCase()]) {
+      goalIdMapping(event, goalListMap, message);
+    }
+
+    // Mapping Standard Events
+    if (standardEventsMap[event.toLowerCase()]) {
+      standardEventsMapping(event, standardEventsMap, message);
+    }
+    // Mapping Ecommerce Events
+    const trimmedEvent = event.toLowerCase().trim();
```

Comment:
Why are we not trimming before standard event mapping?

Reply:
Added